python/map_closures/pipeline.py

```diff
@@ -97,7 +97,7 @@ def __init__(
         else:
             self._eval = False
```

Comment:
I would add `self.results = None` here to avoid a repeated check for `if self.eval` below